src/DIRAC/ConfigurationSystem/Client/Utilities.py

```diff
@@ -578,3 +569,41 @@ def getDIRACGOCDictionary():
 
   log.debug('End function.')
   return S_OK(dictionary)
+
+
+def getAuthAPI():
+  """ Get Auth REST API url
+
+      :return: str
+  """
+  return gConfig.getValue("/Systems/Framework/%s/URLs/AuthAPI" % getSystemInstance("Framework"))
+
+
+def getAuthorizationServerMetadata(issuer=None):
+  """ Get authorization server metadata
+
+      :return: S_OK(dict)/S_ERROR()
+  """
+  result = gConfig.getOptionsDictRecursively('/DIRAC/Authorization')
+  if not result['OK']:
+    return {'issuer': issuer} if issuer else result
+  data = result['Value']
+
+  # Research DIRAC Authorization Server issuer
+  data['issuer'] = data.get('issuer', issuer)
+  if not data['issuer']:
+    try:
+      data['issuer'] = getAuthAPI()
+    except Exception as e:
+      return S_ERROR('No issuer found in DIRAC authorization server: %s' % repr(e))
```

Comment:
It does not look like `getAuthAPI` may issue exceptions. Instead it might well return an empty string.

Reply:
when I use this on the DIRAC client side, where there is no authorization yet and the local configuration contains only the basics, I get an exception.